metricbeat/module/ceph/command.go

```diff
@@ -0,0 +1,45 @@
+package ceph
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+func Execute(c *CephConfig, command string) (string, error) {
+
+	cmdArgs := []string{}
+	cmdArgsDefault := []string{"--conf", c.ConfigPath, "--name", c.User, "--format", "json"}
+	cmdArgs = append(cmdArgs, cmdArgsDefault...)
+	cmdArgs = append(cmdArgs, strings.Split(command, " ")...)
+
+	var output string
+
+	if strings.Contains(c.BinaryPath, "docker") {
+		cmdCeph := []string{"/usr/bin/ceph"}
```

Comment:
We should not execute binaries whenever possible. As far as I can see ceph also has a REST API to fetch monitoring data: http://ceph.com/planet/experimenting-with-the-ceph-rest-api/ This should be used to fetch the data. Calling a binary requires CEPH to be installed on the machine. This works in most cases but breaks for example in docker environments, where on container is used to monitor the other containers.